### PR TITLE
Breaking change for razor

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to https://github.com/aspnet/Announcements/issues/459. Related to dotnet/aspnetcore.docs#22010.

[Internal preview](https://review.docs.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/6.0/razor-compiler-doesnt-produce-views-assembly?branch=pr-en-us-23720).

### CODE REVIEW

File: docs/core/compatibility/aspnet-core/6.0/razor-compiler-doesnt-produce-views-assembly.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+title: "Breaking change: Razor: Compiler now produces a single assembly"
```
Comment (Rick-Anderson) [inline docs/core/compatibility/aspnet-core/6.0/razor-compiler-doesnt-produce-views-assembly.md:2]: Is there an advantage to using `"`, quotes, in title/description? We never use them. I know it doesn't hurt.

File: docs/core/compatibility/aspnet-core/6.0/razor-compiler-doesnt-produce-views-assembly.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+- A main *AppName.dll* assembly that contains application types.
+- An *AppName.Views.dll* assembly that contains the generated views that are defined in the app. Generated view types are `public` and under the `AspNetCore` namespace.
+
+## New behavior
+
+Both views and application types are included in a single *AppName.dll* assembly. View types have the accessibility modifiers `internal` and `sealed`, by default, and are included under the `AspNetCoreGeneratedDocument` namespace.
+
+## Reason for change
+
+Removing the two-step compilation process improves build performance for applications that use Razor views. This change also allows Razor views to participate in the "hot reload" experience for Visual Studio.
```
Comment (Rick-Anderson) [inline docs/core/compatibility/aspnet-core/6.0/razor-compiler-doesnt-produce-views-assembly.md:28]:
```suggestion
Removing the two-step compilation process:

 * Improves build performance for applications that use Razor views.
 * Allows Razor views to participate in the "hot reload" experience for Visual Studio.
```

**Review by Rick-Anderson — APPROVED** [review]

File: docs/core/compatibility/aspnet-core/6.0/razor-compiler-doesnt-produce-views-assembly.md (hunk 1 above)
Comment (tdykstra) [reply docs/core/compatibility/aspnet-core/6.0/razor-compiler-doesnt-produce-views-assembly.md:2]: You have to use quotes if the text contains colons.

**Comment by Rick-Anderson** [comment]
There are more changes too. I was planning on putting them in the [Migrate from ASP.NET Core 5.0 to 6.0](https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60?view=aspnetcore-6.0&tabs=visual-studio). One I do that, I'll update this doc.  Here's an outline

## Changes to the ASP.NET Core Razor SDK

The Razor compiler in .NET 6 leverages the new source generators feature to generated compiled C# files from the Razor views and pages in a document. In previous versions:

* The compilation relied on an individual `RazorGenerate` and `RazorCompile` target to produce the generated code. These targets are no longer valid. In .NET 6, both code generation and compilation are supported by a single call to the compiler.
* A separate Razor assembly was generated that contained the compiled view types in an application. This behavior has been deprecated and a single assembly is produced that contains both the app types and the generated views.

- Remove all the targets listed under `Targets`. Users can continue to use `RazorComponentGenerateDependsOn` to do things that need to happen before the build runs.
- Under "Properties" add the `EmitCompilerGeneratedFiles` property to the table. For description, we can say "When set to `true`, it writes the generated source files to disk. Useful when debugging buggy behavior in the compiler. `false` by default."
- Under "Properties", remove the following properties from the table as they are no longer applicable with the single-step compilation model.
  - `RazorTargetAssemblyAttribute`
  - `RazorTargetName`
  - `EnableDefaultRazorTargetAssemblyInfoAttributes`
  - `UseRazorBuildServer`

File: docs/core/compatibility/aspnet-core/6.0/razor-compiler-doesnt-produce-views-assembly.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+no-loc: [ Razor ]
+ms.date: 04/08/2021
+---
+# Razor: Compiler no longer produces a Views assembly
+
+The Razor compiler no longer produces a separate *Views.dll* file that contains the CSHTML views defined in an application.
+
+## Version introduced
+
+ASP.NET Core 6.0 Preview 3
```
Comment (Rick-Anderson) [inline docs/core/compatibility/aspnet-core/6.0/razor-compiler-doesnt-produce-views-assembly.md:13]:
```suggestion
ASP.NET Core 6.0 Preview 2
```

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

**Review by Rick-Anderson — APPROVED** [review]

File: docs/core/compatibility/aspnet-core/6.0/razor-compiler-doesnt-produce-views-assembly.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+## Old behavior
+
+In previous versions, the Razor compiler utilizes a two-step compilation process that produces two files:
+
+- A main *AppName.dll* assembly that contains application types.
```
Comment (gewarren) [inline docs/core/compatibility/aspnet-core/6.0/razor-compiler-doesnt-produce-views-assembly.md:19]:
```suggestion
- A main *AppName.dll* assembly that contained application types.
```

File: docs/core/compatibility/aspnet-core/6.0/razor-compiler-doesnt-produce-views-assembly.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+The Razor compiler no longer produces a separate *Views.dll* file that contains the CSHTML views defined in an application.
+
+## Version introduced
+
+ASP.NET Core 6.0 Preview 3
+
+## Old behavior
+
+In previous versions, the Razor compiler utilizes a two-step compilation process that produces two files:
```
[the remark this answers is not in the document]
Comment (gewarren) [reply docs/core/compatibility/aspnet-core/6.0/razor-compiler-doesnt-produce-views-assembly.md:17]: If I accept this change, then I need to make the two sentences about the assemblies past tense too, at which point it doesn't seem to make sense anymore. If you use an older version, then it's still doing that behavior in the present tense.

**Pull request merged**